library/extractor/src/main/java/com/google/android/exoplayer2/extractor/avi/VideoFormat.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.exoplayer2.extractor.avi;
+
+import androidx.annotation.VisibleForTesting;
+import com.google.android.exoplayer2.util.MimeTypes;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+
+/**
+ * Wrapper around the BITMAPINFOHEADER structure
+ */
+public class VideoFormat {
+
+  static final int XVID = 'X' | ('V' << 8) | ('I' << 16) | ('D' << 24);
+
+  private static final HashMap<Integer, String> STREAM_MAP = new HashMap<>();
+
+  static {
+    //Although other types are technically supported, AVI is almost exclusively MP4V and MJPEG
+    final String mimeType = MimeTypes.VIDEO_MP4V;
+    //final String mimeType = MimeTypes.VIDEO_H263;
+
+    //I've never seen an Android devices that actually supports MP42
+    STREAM_MAP.put('M' | ('P' << 8) | ('4' << 16) | ('2' << 24), MimeTypes.BASE_TYPE_VIDEO+"/mp42");
+    //Samsung seems to support the rare MP43.
+    STREAM_MAP.put('M' | ('P' << 8) | ('4' << 16) | ('3' << 24), MimeTypes.BASE_TYPE_VIDEO+"/mp43");
```

Comment:
I would like to challenge that.  I think this syntax greatly improves the readability and understandability of the code.  Further, the compiler converts these to constants, so there is no runtime overhead.

I completely agree on the mime types constants.  My bad.

Reply:
Fixed the MimeTypes, still would like to discuss the constants.

Reply:
I agree it is more readable. And I agree it's likely that the compiler will fold constants.

However, even more readable than tis was the approach used before https://github.com/google/ExoPlayer/commit/21be28431840f8adf73ea1cbed3e2c7fa7cf86c7, where you'd have `Util.getIntegerCodeForString("34pm")` (or similar). To allow inlining, hex constants where adopted instead. You can rightfully argue that operating over chars is more readable, but someone in the internal review will most likely argue that this goes against what we do everywhere else (see for instance AtomParsers.java, or Atom.java), and usually consistency takes precedence.

It's up to you, I can make this change when I merge this otherwise.

Reply:
> I think the most important part right now is that we avoid the unbounded int array, which is adding a lot of code without lots of benefit. Otherwise, I think I'll be able to merge some time soon.

I covered this in detail above, but conceptually, I think it's required.  It would be possible to inline the code into readIdx(), but I think that makes an already complex method more complex.  I prefer to write small reusable objects over monolithic methods.  It makes the testing much easier.  It also may be possible to replace it with a List<> of some variety.

Reply:
The Util.getIntegerCodeForString() method did not allow the constants to be used in switch{} statements!  I AssUMe it was because they were getting resolved at run time.

Just because you have always done things one way, does not mean it's the best way.  :)  I would challenge you to bring this (or another) approach up to the team.  Writing char values as hex values in reverse order is clunky and highly error prone.

That said a non-optimal standard is better than no standard.  I'll make the update.

Reply:
Done, but I cried a little.